### ISSUE TITLE
Carbon Gantt minimum chunk width not honored unless start/end dates are identical

### ISSUE THREAD

**Issue by Matthematic** [cerner/terra-graphs#141]
# Bug Report

## Description
We have received a complaint from a client about tasks being drawn extremely skinny:
![image](https://user-images.githubusercontent.com/6292111/120359111-ecfc3280-c2cc-11eb-90ea-1aee04d86332.png)

Looking into this, I found that there is no problem with the data that is generating that task - it is technically correct - but that we need to support a minimum width on tasks to prevent that situation. Currently, Gantt supports a "chunk" task which I believe was intended to enforce a minimum width on tasks. However this only works if the start and end dates are identical, which I believe can be considered a bug since a truly robust enforcement of minimum width would have to compare to the start/end dates of the graph itself to determine expected size. 

### Graphs affected

* [ ]  Bar graph
* [ ]  Bubble graph
* [X]  Gantt chart
* [ ]  Line graph
* [ ]  Paired result
* [ ]  Pie chart
* [ ]  Scatter plot
* [ ]  Timeline

## Steps to Reproduce

1. Render a Gantt graph
2. Draw a task with start and end dates only 1 minute apart

## Link to CodeSandbox
https://codesandbox.io/s/empty-fast-ch35j

## Expected Behavior

I would expect the same minimum chunk width to get applied in this scenario

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by lj017464** [comment]
@mallorymcc looking for an update on this issue, so that we can update the client. We have a reassembly correction JIRA logged on our end https://jira2.cerner.com/browse/CCDEV-16545 that we are tracking on.

**Comment by mraplumb** [comment]
@lj017464 and @Matthematic We have been discussing this request and we are not planning to make any changes given concerns about the requested changes.  Duration of time is an essential element to tasks on a Gantt chart and when a task has a start date/time and end date/time that are not the same, we want to make sure that that duration of time is represented in the graph.  When the start and end date/times are the same, we create the chunk since we know time is not a factor.

Is the representation of time not essential to the tracks on your graph where you are wanting to increase the size?  If not, can you consider making the date/times the same to get the chunk size?  If duration is important, would the events concept for Gantt charts be something that is considered?

**Comment by lj017464** [comment]
Yes, the representation of time is essential in our component and carbon tasks work great for us when the start and end times are the same, or when a long duration is specified:
![image](https://user-images.githubusercontent.com/86629146/124309422-4fe12380-db30-11eb-9327-53c79e04d255.png)

However, tasks with a short duration are hardly recognizable in our component, which is what the client is raising as a concern: 
![image](https://user-images.githubusercontent.com/86629146/124309442-566f9b00-db30-11eb-8727-85cab67213ac.png)

Due to this, we are asking for the ability to set a minimum width for a task, such that a task cannot be less than the size of the chunk. 

Would Carbon entertain adding some type of consumer configuration to allow for this?

**Comment by mraplumb** [comment]
@lj017464 can you provide a full example that shows what this looks like with real data for your concept and in the context of the rest of your application/view?

One of our biggest concerns is having non-equal weighted time boxes on the Gantt chart especially if it involves clinical data.  In order to make an informed decision we will need to see this in context and get input from our UX teams.

**Comment by lj017464** [comment]
I've attached a few examples provided by the client:
![image](https://user-images.githubusercontent.com/86629146/125676430-cbc2c85d-5ad5-465e-9b6d-538e314206f6.png)
![image](https://user-images.githubusercontent.com/86629146/125676477-9b4f60ba-6a12-4bbf-bb5d-a1a671750e32.png)

**Comment by lj017464** [comment]
I've attached a few examples provided by the client -- see the gantt display for the ceFAZolin order (first order)
![image](https://user-images.githubusercontent.com/86629146/125676690-99a4beae-a85c-485d-9607-6a554146838a.png)

**Comment by lj017464** [comment]
![image](https://user-images.githubusercontent.com/86629146/125676918-47e05737-46bb-4490-ae5a-86f7ababc16b.png)

**Comment by lj017464** [comment]
The ceFAZolin administration had an infuse over period of 3 minutes. This is being viewed in a 48 hour view in our component.

**Comment by lj017464** [comment]
Hi @mraplumb -- I'm being asked by the engagement owner for an update on this particular issue. I mentioned that the UX foundations team has been engaged and that you were having further discussions on this. Are there any updates from your discussions as of last week?

**Comment by mraplumb** [comment]
@lj017464 I am following up with the UX team to see if there are any further updates/input from their review of what was posted.

**Comment by kc4181** [comment]
Meeting notes from 8/2:

Concerns about keeping time intervals consistent and what setting a minimum size might to do that
- Resolution needs to keep in mind all consumers of the Gantt chart and ensure it doesn't break other use cases
- Durations need to be kept consistent in terms of scale to ensure they are not misinterpreted if comparing one to another
- Needs to also work when Carbon is displaying with Stella timeline
  - Making the tasks render wider/larger with a minimum width can make them appear that they occurred at a time which is not accurate

UX will investigate visual updates that would more clearly distinguish the smaller time intervals on the graphs
- Includes investigating different visual styling for items that are event-based vs time-based

Will meet again on 8/9 to review UX investigation

**Comment by jeremyfuksa** [comment]
**UX Foundations Update:** We have worked on this for the past week and have shared our thoughts with Justin Peterson. He is adding a couple of his own thoughts and will get in touch outside of this thread soon.

**Comment by chrismichalewicz** [comment]
@jeremyfuksa @mraplumb Testing your memory/notes on something about a year old, do you  know where this issue landed?  Is there still outstanding items for the dev team to consider?

**Comment by Matthematic** [comment]
@chrismichalewicz I don't recall ever hearing anything regarding this issue. I believe I am still waiting to see if this is something that this project can support or if we need to support it by fudging with the dates on our end.
